cluster-provision/okd/scripts/run.sh

```diff
@@ -95,10 +115,39 @@ if [[ ${worker_node_ip} != "192.168.126.51" ]]; then
     haproxy -f /etc/haproxy/haproxy.cfg
 fi
 
-until [[ $(oc get pods --all-namespaces --no-headers | grep -v revision-pruner | grep -v Running | grep -v Completed | wc -l) -le 3 ]]; do
-    echo "waiting for pods to come online"
+set +xe
+n=0
+until [[ $(false) ]]; do
+    oc get pods --all-namespaces --no-headers > /tmp/out.txt
+    if [ $? -eq 0 ]; then
+        NUM_PODS=$(cat /tmp/out.txt | grep -v revision-pruner | grep -v Running | grep -v Completed | wc -l)
```

Comment:
This whole `until` contains some logic that is not clear on the first sight. Please provide inline comments